include/cubos/settings.hpp

```diff
@@ -0,0 +1,63 @@
+#ifndef CUBOS_SETTINGS_HPP
+#define CUBOS_SETTINGS_HPP
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+namespace cubos
+{
+
+    class Settings
+    {
+    protected:
+        Settings()
+        {
+        }
+        static Settings* _instance;
+
+    public:
+        Settings(Settings& settings) = delete;
+        void operator=(const Settings&) = delete;
+        static Settings* GetInstance();
+
+        void clear();
+
+        void setString(const std::string& key, const std::string& value);
+        std::string getString(const std::string& key, const std::string& defaultValue);
+
+    private:
+        std::unordered_map<std::string, std::string> _values;
+    };
+
+} // namespace cubos
+
+cubos::Settings* cubos::Settings::_instance = nullptr;

```

Comment:
This will cause a compilation error when the header is included in more than one compilation unit, since there will be a duplicate definition. This can moving be easily solved by moving the definitions to a `src/settings.cpp` file.

Reply:
Added to settings.cpp